integrations/wpseo/wpseo.php

```diff
@@ -127,40 +127,43 @@ protected function wpseo_get_active_languages() {
 	}
 
 	/**
-	 * Modifies the sql request for posts sitemaps
-	 * Only when using multiple domains or subdomains or if some languages are not active
+	 * Modifies the sql request for posts sitemaps.
 	 *
 	 * @since 1.6.4
 	 *
-	 * @param string $sql       JOIN clause
-	 * @param string $post_type
+	 * @param string $sql       JOIN clause.
+	 * @param string $post_type Post type.
 	 * @return string
 	 */
 	public function wpseo_posts_join( $sql, $post_type ) {
-		return pll_is_translated_post_type( $post_type ) && ( PLL()->options['force_lang'] > 1 || $this->wpseo_get_active_languages() ) ? $sql . PLL()->model->post->join_clause() : $sql;
+		return pll_is_translated_post_type( $post_type ) ? $sql . PLL()->model->post->join_clause() : $sql;
 	}
 
 	/**
-	 * Modifies the sql request for posts sitemaps
-	 * Only when using multiple domains or subdomains or if some languages are not active
+	 * Modifies the sql request for posts sitemaps.
 	 *
 	 * @since 1.6.4
 	 *
-	 * @param string $sql       WHERE clause
-	 * @param string $post_type
+	 * @param string $sql       WHERE clause.
+	 * @param string $post_type Post type.
 	 * @return string
 	 */
 	public function wpseo_posts_where( $sql, $post_type ) {
-		if ( pll_is_translated_post_type( $post_type ) ) {
-			if ( PLL()->options['force_lang'] > 1 ) {
-				return $sql . PLL()->model->post->where_clause( PLL()->curlang );
-			}
+		if ( ! pll_is_translated_post_type( $post_type ) ) {
+			return $sql;
+		}
 
-			if ( $languages = $this->wpseo_get_active_languages() ) {
-				return $sql . PLL()->model->post->where_clause( $languages );
-			}
+		if ( PLL()->options['force_lang'] > 1 ) {
+			return $sql . PLL()->model->post->where_clause( PLL()->curlang );
```

Comment:
Shouldn't we test `PLL()->curlang` is an instance of `PLL_Language` and not `false` or `null` before calling `where_clause()`?

Reply:
+1, since we are editing this part...
PhpStan reports the issue:
```
Parameter #1 $lang of method PLL_Translated_Object::where_clause() expects array<string>|PLL_Language|string, PLL_Language|false|null given.
```

Reply:
Done in [1beb0d1](https://github.com/polylang/polylang/pull/1103/commits/1beb0d1d684463d75c6297c0c46452024a72cd0e).